cloudify_cli/commands/plugins.py

```diff
@@ -516,18 +522,57 @@ def update(blueprint_id,
             'only the specific plugins, use --plugin-name parameter instead.')
 
     utils.explicit_tenant_name_message(tenant_name, logger)
+    mappings = _load_mapping_file(mapping_file) if mapping_file else {}
     if blueprint_id:
         _update_a_blueprint(blueprint_id, plugin_names,
                             to_latest, all_to_latest, to_minor, all_to_minor,
+                            mappings.get(blueprint_id),
                             include_logs, json_output, logger,
                             client, force)
     elif all_blueprints:
-        for blueprint in client.blueprints.list():
-            _update_a_blueprint(blueprint.id, plugin_names,
-                                to_latest, all_to_latest,
-                                to_minor, all_to_minor,
-                                include_logs, json_output, logger,
-                                client, force)
+        update_results = {'successful': [], 'failed': []}
+        pagination_size = 100
+        pagination_offset = 0
+        while True:
+            blueprints_processed = 0
+            for blueprint in client.blueprints.list(
+                sort='created_at',
+                _offset=pagination_offset,
+                _size=pagination_size,
+            ):
+                try:
+                    _update_a_blueprint(blueprint.id, plugin_names,
+                                        to_latest, all_to_latest,
+                                        to_minor, all_to_minor,
+                                        mappings.get(blueprint.id),
+                                        include_logs, json_output, logger,
+                                        client, force)
+                    update_results['successful'].append(blueprint.id)
+                except CloudifyClientError as ex:
+                    update_results['failed'].append(blueprint.id)
+                    logger.warning('Error during %s blueprint update.  %s',
+                                   blueprint.id, ex)
+                finally:
+                    blueprints_processed += 1
+            pagination_offset += pagination_size
+            if blueprints_processed != pagination_size:
```

Comment:
I think you can actually get a total count from the result, which would be easier to parse?